### PR TITLE
Allow all punctuations to be escaped

### PR DESCRIPTION
**Pull request by Bale001**
Previously, regex's like `/\-/`, or `/\!/` would fail to compile. In javascript (and other ecmascript based languages) these regex's are allowed, so they should be allowed in this crate as well.

For more info see https://github.com/ruffle-rs/ruffle/issues/9859

### CODE REVIEW

**Comment by selfisekai** [comment]
this is a yet another difference between unicode and normal regexes. this behavior should not be removed but gated by `if self.flags.unicode`

**Comment by selfisekai** [comment]
also, add tests here: https://github.com/ridiculousfish/regress/blob/master/tests/tests.rs#L313

**Comment by ridiculousfish** [comment]
Yes, @selfisekai is right. One can verify in node:
```
node
Welcome to Node.js v19.4.0.
Type ".help" for more information.
> /\-/
/\-/
> /\-/u
/\-/u
^^^^^

Uncaught SyntaxError: Invalid regular expression: /\-/: Invalid escape
```

**Comment by Bale001** [comment]
Thanks for letting me know about that.

While testing this, I noticed that it actually allows escaping any character, not just punctuations, so I fixed that

File: tests/tests.rs [hunk 1]
```diff
@@ -311,6 +311,11 @@ fn run_nonunicode_tests() {
 }
 
 fn run_nonunicode_test_tc(tc: TestConfig) {
+    // escaping unrecognised chars
+    tc.compilef(r"\ ", "").match1f(r" ").test_eq(r" ");
+    tc.compilef(r"\ a", "").match1f(r" a").test_eq(r" a");
+    tc.compilef(r"a\ ", "").match1f(r"a ").test_eq(r"a ");
```
Comment (Lord-McSweeney) [inline tests/tests.rs:317]: I think this should also have a test with escapes using the unicode flag

**Comment by ridiculousfish** [comment]
Merged as 603833d4320302f115438e1bf9d37bd30b01fa9b, thank you!
